packages/services/src/index.ts

```diff
@@ -21,30 +22,40 @@ export class Services implements ServicesInterface {
     return this._querier;
   }
 
-  async initialize(): Promise<void> {
-    try {
-      this._querier = new RootQuerier({ grpcEndpoint: this.config.grpcEndpoint });
+  public async initialize(): Promise<void> {
+    this._querier = new RootQuerier({ grpcEndpoint: this.config.grpcEndpoint });
 
-      await this.tryToSync();
-    } catch (e) {
-      // Logging here as service worker does not appear to bubble the errors up
-      console.error(e);
-      throw e;
-    }
+    // await startup for any errors
+    await this.getWalletServices();
+
+    // void sync to background
+    void this.autoRetrySync();
   }
 
-  async tryToSync() {
-    try {
-      const ws = await this.getWalletServices();
-      void ws.blockProcessor.syncBlocks();
-    } catch {
-      // With throw if no wallet. Can ignore.
-    }
+  // TODO: anyone holding a reference to the old walletServices may attempt to use it
+  private async autoRetrySync() {
+    await backOff(
+      () => this.getWalletServices().then(({ blockProcessor }) => blockProcessor.sync()),
+      {
+        maxDelay: 30_000, // 30 seconds
+        retry: (e, attemptNumber) => {
+          // don't interfere with deliberate stop
+          if (typeof e === 'string' && e.startsWith('Sync abort')) return false;
+
+          // backoff will swallow the error, so log it
+          console.warn('Sync failure', e);
+          // next caller will re-init services
+          this.walletServicesPromise = undefined;
+          console.log('Sync retry', attemptNumber);
+          return true;
+        },
+      },
+    );
```

Comment:
this is the relocated retry logic. anything that isn't a deliberate abort will completely re-init services. it is possible that some in-flight request may be holding a reference to the old services, and unfortunately it may still be possible to use them

Reply:
Hopefully that would only be our rpc methods which can be refetched and the services get re-requested